### PR TITLE
Update py2exe to 0.13.0.2

### PR DESCRIPTION
**Pull request by dpy013**
-->

### Link to issue number:
[Release Notes](https://github.com/py2exe/py2exe/releases/tag/v0.13.0.2)
### Summary of the issue:
Updating py2exe to 0.13.0.2
This is a bugfix release, check the release link above for the exact changelog!
### Description of user facing changes
There are no user-side changes, this is an update to the build component
### Description of development approach
Updated the py2exe version of the requirements.txt file
and did a local build with scons launcher version=py2exe-test
### Testing strategy:
Need to see if appveyor can be built successfully
### Known issues with pull request:
Not found at the moment
### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

## Summary by CodeRabbit

- **Chores**
	- Updated the specification of the `py2exe` dependency to a fixed version for improved clarity and more predictable builds.

### CODE REVIEW

**Comment by seanbudd** [comment]
@dpy013 - from the change log there's a few notable things:

- Newer versions of setuptools are now supported. Can you update us to the latest setuptools too?
- @lukaszgo1 made a contribution to fix an issue we were seeing in NVDA, https://github.com/py2exe/py2exe/issues/193. It would be nice to restore that information to our builds now that it is possible again - see changes to setup.py in https://github.com/nvaccess/nvda/pull/15544/files

**Comment by seanbudd** [comment]
Actually my mistake, we already fixed up the version info stuff fixed by Lukasz in https://github.com/nvaccess/nvda/pull/15544/commits/b8124fb180a6cd43760e717dc177c5f4efa91104
However could you still please bump setuptools

File: venvUtils/ensureVenv.py [hunk 1]
```diff
@@ -94,7 +94,7 @@ def populate():
 			"-m",
 			"pip",
 			"install",
-			"setuptools==69.5.1",
+			"setuptools==70",
```
Comment (seanbudd) [inline venvUtils/ensureVenv.py:97]: setuptools is also mentioned in pyproject.toml, please ensure this is updated to reference the latest setuptools.

The latest setuptools is 72, have you tested this version? if so, what didn't work.

Please fixup the comment above which references py2exe and setuptools

File: user_docs/en/changes.md [hunk 2]
```diff
@@ -22,6 +22,13 @@
 * In on-demand speech mode, NVDA does not talk anymore when a message is opened in Outlook, when a new page is loaded in a browser or during the slideshow in PowerPoint. (#16825, @CyrilleB79)
 * In Mozilla Firefox, moving the mouse over text before or after a link now reliably reports the text. (#15990, @jcsteh)
 
+### Changes
+
+* Component updates:
+
+  * Updated py2exe to 0.13.0.2 (#16907, @dpy013)
+
+  * re enable setuptools70
```
Comment (seanbudd) [inline user_docs/en/changes.md:31]:
```suggestion
* Component updates:
  * Updated py2exe to 0.13.0.2 (#16907, @dpy013)
  * Updated setuptools to 72.0.0 (#16907, @dpy013)
```

File: requirements.txt [hunk 3]
```diff
@@ -20,7 +20,7 @@ typing-extensions==4.9.0
 pycaw==20240210
 
 # Packaging NVDA
-py2exe @ git+https://github.com/py2exe/py2exe@4e7b2b2c60face592e67cb1bc935172a20fa371d#egg=py2exe
+py2exe ==0.13.0.2
```
Comment (seanbudd) [inline requirements.txt:23]:
```suggestion
py2exe==0.13.0.2
```

File: venvUtils/ensureVenv.py [hunk 4]
```diff
@@ -94,7 +94,7 @@ def populate():
 			"-m",
 			"pip",
 			"install",
-			"setuptools==70",
+			"setuptools==72.0.0",
```
Comment (seanbudd) [inline venvUtils/ensureVenv.py:97]: I think we want to match any 72 subversion 

```suggestion
			"setuptools~= 72",
```

File: pyproject.toml [hunk 5]
```diff
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools<70", "wheel"]
+requires = ["setuptools<72.0.0", "wheel"]
```
Comment (seanbudd) [inline pyproject.toml:2]: We don't want to pin less than 72, we want to use 72

```suggestion
requires = ["setuptools~=72", "wheel"]
```

File: user_docs/en/changes.md [hunk 6]
```diff
@@ -22,6 +22,13 @@
 * In on-demand speech mode, NVDA does not talk anymore when a message is opened in Outlook, when a new page is loaded in a browser or during the slideshow in PowerPoint. (#16825, @CyrilleB79)
 * In Mozilla Firefox, moving the mouse over text before or after a link now reliably reports the text. (#15990, @jcsteh)
 
+### Changes
+
+* Component updates:
+
+  * Updated py2exe to 0.13.0.2 (#16907, @dpy013)
+
+  * Updated setuptools to 72.0.0 (#16907, @dpy013)
```
Comment (seanbudd) [inline user_docs/en/changes.md:31]:
```suggestion
* Component updates:
  * Updated py2exe to 0.13.0.2 (#16907, @dpy013)
  * Updated setuptools to 72.0.0 (#16907, @dpy013)
```

File: venvUtils/ensureVenv.py (hunk 4 above)
Comment (seanbudd) [reply venvUtils/ensureVenv.py:97]: As requested previously, please fixup the code comment above this line which references py2exe and setuptools

File: pyproject.toml (hunk 5 above)
Comment (dpy013) [reply pyproject.toml]: Local compilation has been installed correctly
Collecting setuptools==72
e8/setuptools-72.0.0-py3-none-any.whl (2.3 MB)

**Comment by dpy013** [comment]
Hello, I've reorganized the section on setuptools to
Uniformly assign setuptools to the modified version of the comment: 72.0.
Instead of the original 72.0.0.
Now setuptools installed by running scons source will select from the 72.0 major version, e.g. the original 72.0.0.
Also re-optimized changes
Please let me know if you have any more questions.

File: pyproject.toml (hunk 5 above)
Comment (seanbudd) [reply pyproject.toml]: Doesn't matter. Please update pyproject.toml to match setup.py.
They should refer to the same setuptools version

File: venvUtils/ensureVenv.py [hunk 7]
```diff
@@ -94,7 +94,7 @@ def populate():
 			"-m",
 			"pip",
 			"install",
-			"setuptools==69.5.1",
+			"setuptools==72.0",
```
Comment (seanbudd) [inline venvUtils/ensureVenv.py:97]:
```suggestion
			"setuptools~=72.0",
```

File: pyproject.toml [hunk 8]
```diff
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools<70", "wheel"]
+requires = ["setuptools<72.0", "wheel"]
```
Comment (seanbudd) [inline pyproject.toml:2]:
```suggestion
requires = ["setuptools~=72.0", "wheel"]
```

File: venvUtils/ensureVenv.py (hunk 7 above)
Comment (seanbudd) [reply venvUtils/ensureVenv.py:97]: As requested twice previously, please fixup the code comment above this line which references py2exe and setuptools

```
			# py2exe is not compatible with setuptools 70+
			# wheel must be manually installed when creating an non-isolated build with a custom setuptools version.
```

should be replaced with

```
			# wheel must be manually installed when creating an non-isolated build with a custom setuptools version.
```

File: user_docs/en/changes.md [hunk 9]
```diff
@@ -27,6 +27,11 @@ The available options are:
 * In on-demand speech mode, NVDA does not talk anymore when a message is opened in Outlook, when a new page is loaded in a browser or during the slideshow in PowerPoint. (#16825, @CyrilleB79)
 * In Mozilla Firefox, moving the mouse over text before or after a link now reliably reports the text. (#15990, @jcsteh)
 
+### Changes
+
+* Component updates:
+  * Updated py2exe to 0.13.0.2 (#16907, @dpy013)
+  * Updated setuptools to 72.0 (#16907, @dpy013)
```
Comment (seanbudd) [inline user_docs/en/changes.md:34]: This section should be below "New Features" I'm not sure why/how it was missed when starting this release

File: venvUtils/ensureVenv.py [hunk 10]
```diff
@@ -88,13 +88,13 @@ def populate():
 			"--upgrade",
 			"pip",
 			"&&",
-			# py2exe is not compatible with setuptools 70+
-			# wheel must be manually installed when creating an non-isolated build with a custom setuptools version.
+			# py2exe is now compatible with setuptools 72.0
+			# There is now no need to manually install the wheel when creating non-isolated builds using a customized version of setuptools.
```
Comment (seanbudd) [inline venvUtils/ensureVenv.py:92]: `wheel` is a package installed separately to setuptools. This comment should not be changed. Please listen to suggestions. The previous suggestion just asked you to remove the line for setuptools

```suggestion
			# wheel must be manually installed when creating an non-isolated build with a custom setuptools version.
```

**Review by seanbudd — APPROVED** [review]

**Review by seanbudd — APPROVED** [review]

**Pull request merged**